crates/bevy_asset/src/folder.rs

```diff
@@ -1,8 +1,64 @@
-use alloc::vec::Vec;
+use alloc::{sync::Arc, vec::Vec};
 
 use crate::{Asset, UntypedHandle};
 use bevy_reflect::TypePath;
 
+/// This is use for [`AssetServer::load_folder_with_filter`](crate::prelude::AssetServer::load_folder_with_filter).
+#[derive(Debug, Clone, Copy, Default)]
+pub enum LoadFilterKind {
+    #[default]
+    White, //Allow loading
```

Comment:
We should rename these to `Allow` / `Deny`, and document what the variants do via doc comments :)